src/zod/userSignup.schema.ts

```diff
@@ -0,0 +1,21 @@
+import { z, ZodType } from 'zod';
+import { SignupFormData } from '#/modules/auth/types';
+
+export const UserSignupSchema: ZodType<SignupFormData> = z
+  .object({
+    firstName: z
+      .string()
+      .min(1, { message: 'Please enter your first name' })
+      .max(64, { message: 'Entered first name is too long' }),
+    lastName: z.string().min(1, { message: 'Please enter your last name' }).max(64, { message: 'Entered last name is too long' }),
```

Comment:
Max length for firstName and lastName is 255 in the backend/database. So you should not limit it to 64 here.